### PR TITLE
ANDROAPP-5465-mobile-ui-Button-Block

### PR DESCRIPTION
**Pull request by xavimolloy**

### CODE REVIEW

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ButtonBlock.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * DHIS2 ButtonBlock with generic buttons slot.
+ * @param primaryButton Controls first or primary button, if there is
+ * only one it will be centered, otherwise spaced equally
+ * @param secondaryButton Controls the second button to be shown
+ * @param modifier allow a modifier to be passed to the composable.
+ */
+@Composable
+fun ButtonBlock(
+    primaryButton: @Composable (() -> Unit)? = null,
```
Comment (DavidAparicioAlbaAsenjo) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ButtonBlock.kt:21]: why primary button is nullable?

Comment (xavimolloy) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ButtonBlock.kt]: corrected

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/Button.kt [hunk 2]
```diff
@@ -171,10 +208,10 @@ private fun SimpleButton(
         modifier = modifier,
         enabled = enabled,
         colors = buttonColors,
-        shape = ButtonDefaults.outlinedShape,
+        shape = RoundedCornerShape(Radius.Full),
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/Button.kt:211]: Why don't use our custom shapes: `Shape.Full` will match with the shape that you are creating.

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/theme/Shadow.kt [hunk 3]
```diff
@@ -56,12 +58,12 @@ internal fun Modifier.buttonShadow(
             val paint = Paint()
             val frameworkPaint = paint.asFrameworkPaint()
             val spreadPixel = if (hasIcon) Spacing.Spacing0.toPx() else Spacing.Spacing1_5.toPx()
-            val leftPixel = DEFAULT_PADDING
+            val leftPixel = 0f
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/theme/Shadow.kt:61]: is there any Spacing value matching 0f?

Comment (xavimolloy) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/theme/Shadow.kt:61]: there is a Spacing value of 0.dp not without the dp conversion, as pixels are needed for this. I am going to fix all this with the Refactor Spacings issue once the current PRs have been merged

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/Button.kt (hunk 2 above)
Comment (xavimolloy) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/Button.kt:211]: Good idea!

**Review by andresmr — APPROVED** [review]

**Review by Balcan — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Review by Balcan — APPROVED** [review]

**Pull request merged**